nautobot/extras/models/datasources.py

```diff
@@ -167,3 +174,55 @@ def sync(self, user, dry_run=False):
         if dry_run:
             return enqueue_git_repository_diff_origin_and_local(self, user)
         return enqueue_pull_git_repository_and_refresh_data(self, user)
+
+    @contextmanager
+    def clone(self, path=None, branch=None, head=None, shallow=True):
+        """
+        Perform a (shallow) clone of the Git repository in a temporary directory.
+
+        Args:
+            path (str, optional): The absolute directory path to clone into. If not specified, `tempfile.gettempdir()` will be used.
+            branch (str, optional): The branch to checkout. If not set, the GitRepository.branch will be used.
+            head (str, optional): Git commit hash to check out instead of pulling branch latest.
+            shallow (bool, optional): If set, perform a shallow clone.
+
+        Returns:
+            Return the absolute path of the cloned repo if clone was successful
+        """
+
+        try:
+            if branch and head:
+                raise ValueError("Cannot specify both branch and head")
+
+            try:
+                path_name = tempfile.mkdtemp(dir=path)
```

Comment:
```suggestion
                path_name = tempfile.mkdtemp(dir=path, prefix=self.slug)
```

So that the directory name isn't completely random.